atlp-rwanda/rca-phantom-team2-bn
# #9 Create roles and set permissions

#### What does this PR do?

Create roles and set permissions

#### Description of Task to be completed?

Enabling to creation the roles and set their permissions, so that different users can only access & perform data/operations to which they are authorized.

#### How should this be manually tested?

These are the steps to followed while manually testing this PR:

1. Clone the repository and checkout this branch
2. Create `.env` file in the root directory and add variables. Database variables and credentials can be found in `Dockerfile` under `src/db` directory. The database host is **localhost** or **127.0.0.1**
3. In the project root directory, open `docker-compose.yml`
4. Edit the `backend` service and comment the `profiles` section to allow the service to start. Alternatively, you can manually start the container after completing the next step.
5. Run  `sudo docker compose up`  to start the database (postgres), backend and adminer tool containers
6. In your browser, open [http://localhost:8080](http://localhost:8080) and input database login credentials from the second step to log in to the database. 
7. In browser, open the endpoint `/api-docs` for documentation and check all endoints with tags **Roles** and **Permissions**

The following endpoints must work:

`POST /permissions`

`GET /permissions`

`GET /permissions/operations`

`GET /permissions/models`

`POST /roles`

`GET /roles`

`GET /roles/role/{roleId}`

`POST /roles/grant-permission`

#### Any background context you want to provide?

Database postgresql and Adminer, the web relational databases adminstration tool are added as containers.
These are other tasks done in this PR:

1. Docker and docker compose setup
9. Added Adminer web database management tool

#### What are the relevant pivotal tracker/Trello stories?

Trello story: [#9](https://trello.com/c/ZPDYcr0b/9-create-roles-and-set-permissions)

#### Screenshots:

![image](https://user-images.githubusercontent.com/50821898/227950330-1f63b0f0-5608-4f57-9418-ab1368a53c67.png)

![image](https://user-images.githubusercontent.com/50821898/227956683-0e1279b7-78ba-4716-a22a-f847427961b2.png)
